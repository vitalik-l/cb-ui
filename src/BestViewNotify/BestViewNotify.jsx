import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const isFullScreen = () => !((screen.width - window.innerWidth) > 5
  || (screen.height - window.innerHeight) > 5);

const launchFullScreen = () => {
  const doc = document.documentElement;
  if (doc.requestFullScreen) {
    doc.requestFullScreen();
  } else if (doc.mozRequestFullScreen) {
    doc.mozRequestFullScreen();
  } else if (doc.webkitRequestFullScreen) {
    doc.webkitRequestFullScreen();
  }
};

class BestViewNotify extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      bestView: isFullScreen(),
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.checkFullScreen);
  }

  componentWillUmnount() {
    window.removeEventListener('resize', this.checkFullScreen);
  }

  checkFullScreen() {
    this.setState({ bestView: isFullScreen() });
  }

  closeNote() {
    this.setState({ bestView: true });
  }

  render() {
    const { className } = this.props;
    const { bestView } = this.state;
    const classes = classNames(
      className,
      {
        'cb-BestViewNotify': !className,
        hide: bestView,
      },
    );

    return (
      <div className={classes}>
        <div>
          Best Viewed in&nbsp;
          <span className="link" onClick={launchFullScreen}>Full Screen HD</span>
        </div>
        <span id="BestViewNotify-close" onClick={this.closeNote}>&#x274c;</span>
      </div>
    );
  }
}

BestViewNotify.propTypes = {
  className: PropTypes.string,
};

export default BestViewNotify;
