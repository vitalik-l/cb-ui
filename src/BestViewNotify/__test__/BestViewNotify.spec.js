import React from 'react';
import { shallow } from 'enzyme';
import BestViewNotify from '../BestViewNotify';

describe('BestViewNotify', () => {
  const wrapper = shallow(<BestViewNotify />);

  it('default', () => {
    expect(wrapper).toMatchSnapshot();
  });

  /**
     * should set the 'test' class name to root element
     */
  it('custom className', () => {
    wrapper.setProps({ className: 'test' });
    expect(wrapper.hasClass('test')).toBeTruthy();
  });

  /**
     * should remove the 'hide' class from root element
     */
  it('set visible', () => {
    wrapper.setState({ bestView: false });
    expect(wrapper.hasClass('hide')).toBeFalsy();
  });

  /**
     * should set the 'hide' class to root element
     */
  it('hide panel on click', () => {
    wrapper.setState({ bestView: false });
    wrapper.find('#BestViewNotify-close').simulate('click');
    expect(wrapper.hasClass('hide')).toBeTruthy();
  });
});
