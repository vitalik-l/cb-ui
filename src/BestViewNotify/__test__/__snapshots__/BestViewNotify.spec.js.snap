// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestViewNotify default 1`] = `
<div
  className="cb-BestViewNotify hide"
>
  <div>
    Best Viewed in 
    <span
      className="link"
      onClick={[Function]}
    >
      Full Screen HD
    </span>
  </div>
  <span
    id="BestViewNotify-close"
    onClick={[Function]}
  >
    ❌
  </span>
</div>
`;
